handlers/sandbox.go

```diff
@@ -0,0 +1,180 @@
+//
+// https://www.gnu.org/licenses/gpl-3.0.en.html
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an “AS IS” BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package handlers
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"memphis-broker/logger"
+	"memphis-broker/models"
+	"memphis-broker/utils"
+	"net/http"
+	"time"
+
+	"github.com/dgrijalva/jwt-go"
+	"github.com/gin-gonic/gin"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+type SandboxHandler struct{}
+
+func (sbh SandboxHandler) Login(c *gin.Context) {
+	var body models.SandboxLoginSchema
+	ok := utils.Validate(c, &body, false, nil)
+	if !ok {
+		return
+	}
+	google_token := body.Google_token
+	claims, err := ValidateGoogleJWT(google_token)
+	if err != nil {
+		logger.Error("ValidateGoogleJWT: " + err.Error())
+		c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+		return
+	}
+	exist, user, err := IsUserExist(claims.Email)
+	if err != nil {
+		logger.Error("IsUserExist: " + err.Error())
+		c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+		return
+	}
+
+	if !exist {
+		user = models.User{
+			ID:              primitive.NewObjectID(),
+			Username:        claims.Email,
+			Password:        claims.FirstName + "." + claims.LastName,
+			HubUsername:     "",
+			HubPassword:     "",
+			UserType:        "management",
+			CreationDate:    time.Now(),
+			AlreadyLoggedIn: false,
+			AvatarId:        1,
+		}
+		_, err = usersCollection.InsertOne(context.TODO(), user)
+		if err != nil {
+			logger.Error("InsertOne: " + err.Error())
+			c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+			return
+		}
+
+	}
+
+	token, refreshToken, err := createTokens(user)
+	if err != nil {
+		logger.Error("createTokens: " + err.Error())
+		c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+		return
+	}
+
+	if err != nil {
+		logger.Error("Login error: " + err.Error())
+		c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+		return
+	}
+	if !user.AlreadyLoggedIn {
+		usersCollection.UpdateOne(context.TODO(),
+			bson.M{"_id": user.ID},
+			bson.M{"$set": bson.M{"already_logged_in": true}},
+		)
+	}
+	domain := ""
+	secure := false
+	c.SetCookie("jwt-refresh-token", refreshToken, configuration.REFRESH_JWT_EXPIRES_IN_MINUTES*60*1000, "/", domain, secure, true)
+	c.IndentedJSON(200, gin.H{
+		"jwt":               token,
+		"expires_in":        configuration.JWT_EXPIRES_IN_MINUTES * 60 * 1000,
+		"user_id":           user.ID,
+		"username":          user.Username,
+		"user_type":         user.UserType,
+		"creation_date":     user.CreationDate,
+		"already_logged_in": user.AlreadyLoggedIn,
+		"avatar_id":         user.AvatarId,
+	})
+}
+
+type GoogleClaims struct {
```

Comment:
if this struct is not used elsewhere it should start with regular first letter and not capital letter.
Also declarations should be in the top of the file